common/src/stack/report-system/command/report/system/tests/test_other_dhcp.py

```diff
@@ -0,0 +1,99 @@
+import struct
+import socket
+import pytest
+import binascii
+from stack import api
+from random import randint
+
+pxe_networks = [net['network'] for net in api.Call('list network', args=['pxe=true'])]
+interfaces = api.Call('list.host.interface', args=['a:frontend'])
+pxe_interfaces = [inter['interface'] for inter in interfaces if inter['network'] in pxe_networks]
+
+
+def build_dhcp_packet():
+
+	# Unique MAC address
+	mac = ''.join('%02x'%randint(0,255) for x in range(6))
+	mac_address = binascii.unhexlify(mac)
+
+	# Ensure a unique transaction id when a DHCP request
+	# is sent back
+	transaction_id = b''
+	for i in range(4):
+		t = randint(0, 255)
+		transaction_id += struct.pack('!B', t)
+			
+	# Build DHCP discover packet in binary form
+	packet = dict(
+		boot_request =  b'\x01',
+		ethernet =  b'\x01',
+	 	mac_length = b'\x06',
+		hops =  b'\x00',
+		trans_id = transaction_id,
+		seconds = b'\x00\x00',
+	 	bootp_flags = b'\x80\x00',
+		client_ip =  b'\x00\x00\x00\x00',
+		your_ip =  b'\x00\x00\x00\x00',
+		server_ip =  b'\x00\x00\x00\x00',
+		relay_ip =  b'\x00\x00\x00\x00',
+		mac_addr = mac_address,
+		mac_padding =  b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00',
+		server_name = b'\x00' * 67,
+		boot_file =  b'\x00' * 125,
+		magic_cookie =  b'\x63\x82\x53\x63',
+		dhcp_option = b'\x35\x01\x01',
+		client_identifier =  b'\x3d\x06' + mac_address,
+		parameter_list = b'\x37\x03\x03\x01\x06',
+		end_option =  b'\xff'
+	)
```

Comment:
In case you weren't aware, you can also build dictionaries directly using `{"boot_request": b'\x01', ... }`.